tokio-codec/tests/codecs.rs

```diff
@@ -55,6 +55,44 @@ fn lines_decoder() {
     assert_eq!(None, codec.decode_eof(buf).unwrap());
 }
 
+#[test]
+fn lines_decoder_max_length() {
+    const MAX_LENGTH: usize = 6;
+
+    let mut codec = LinesCodec::with_max_length(MAX_LENGTH);
+    let buf = &mut BytesMut::new();
+
+    buf.reserve(200);
+    buf.put("line 1 is too long\nline 2\r\nline 3\n\r\n\r");
+
+    assert!(codec.decode(buf).is_err());
+    assert!(codec.decode(buf).is_err());
```

Comment:
This is interesting.  I hadn't considered the notion of continuing to throw errors until we get a line back that is under the length limit.

My original thought was that if you had `blah blah\r\nfoo\r\n` and your limit was, say, 4 characters... you'd get a single error that ended up consuming all the `blah`s, and your next `decode` would give you back `foo`.

I'm not really sure if one is better than the other, but I like seeing an entirely different approach. :D

Reply:
Hmm. The intention behind adding the line limit is to prevent a potential denial-of-service attack where an attacker feeds us a line that never ends, causing this function to never return. Given that, I thought it was important that we always return when we're at the length limit rather than continuing to read a (potentially unbounded) number of bytes?

Reply:
Yeah, no, totally.  Like I said, I just hadn't even considered it... so it's great to see!

Reply:
Yeah, I just wanted to explain the rationale. :)